workspaces/bazaar/plugins/bazaar/src/components/UserSelector/UserSelector.tsx

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import {
+  Entity,
+  stringifyEntityRef,
+  parseEntityRef,
+} from '@backstage/catalog-model';
+import Typography from '@material-ui/core/Typography';
+import Autocomplete, {
+  createFilterOptions,
+} from '@material-ui/lab/Autocomplete';
+import TextField from '@material-ui/core/TextField';
+import { makeStyles } from '@material-ui/core/styles';
+import { Controller, Control } from 'react-hook-form';
+import { FormValues } from '../../types';
+
+type Props = {
+  users: Entity[];
+  disableClearable: boolean;
+  defaultValue: string | null;
+  label: string;
+  name: 'responsible';
+  control: Control<FormValues, object>;
+  rules?: Record<string, any>;
+};
+
+const useStyles = makeStyles({
+  container: { width: '100%', minWidth: '22rem' },
+  autocomplete: { overflow: 'hidden' },
+});
+
+const filter = createFilterOptions<Entity | string>();
+
+export const UserSelector = ({
+  users,
+  disableClearable,
+  defaultValue,
+  label,
+  name,
+  control,
+  rules,
+}: Props) => {
+  const classes = useStyles();
+
+  const getDisplayName = (value: string | Entity) => {
```

Comment:
Can we refactor this to use the `EntityPresentationApi`? It was built for this purpose. Here's an example: https://github.com/backstage/backstage/blob/6cbe4f999c9c8e9e090d3be5a7c3af71d59e507d/plugins/scaffolder/src/components/fields/EntityPicker/EntityPicker.tsx#L78

Reply:
Awesome! I didn't realize that existed - thanks for sharing. I've updated the `UserSelector` and the PR pics.